### ISSUE TITLE
Bad mathjax rendering

### ISSUE THREAD

**Issue by memeplex** [facelessuser/pymdown-extensions#1342]
With this configuration:

`pymdown-extensions==8.2`

`https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js`

```javascript
MathJax.Hub.Config({
  config: ["MMLorHTML.js"],
  jax: ["input/TeX", "output/HTML-CSS", "output/NativeMML"],
  extensions: ["MathMenu.js", "MathZoom.js"]
});
```

All extension settings set to default values.

I get the following rendering of:

```
When \(a \ne 0\), there are two solutions to \(ax^2 + bx + c = 0\) and they are
$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$
```

![image](https://user-images.githubusercontent.com/2845433/119246417-cd8c3980-bb57-11eb-900d-b370f1d1ad06.png)

This example is taken from MathJax documentation. I think I've correctly followed pymdown-extensions documentation regarding using the default mode instead of generic. The problem disappears when adding a blank line before the display math equation like this:

```
When \(a \ne 0\), there are two solutions to \(ax^2 + bx + c = 0\) and they are

$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$
```

But I believe it should work without the blank line too.

**Comment by memeplex** [comment]
Ahhh 🤦 :

> A block also must contain no empty lines and should be both preceded and followed by an empty line.

But given that [an example](http://docs.mathjax.org/en/latest/basic/mathematics.html#tex-and-latex-input) without preceding blank line was given in the official mathjax documentation, shouldn't it work? One could say that markdown syntax is different in this regard, but I'm unconvinced.

**Comment by facelessuser** [comment]
> A block also must contain no empty lines and should be both preceded and followed by an empty line.

Bingo

> But given that [an example](http://docs.mathjax.org/en/latest/basic/mathematics.html#tex-and-latex-input) without preceding blank line was given in the official mathjax documentation, shouldn't it work?

Okay, so if you are constructing HTML yourself, and using their text parser, then yes. But you aren't. You are using our Markdown parser, which is based on how Python Markdown recognizes blocks. When you don't have a space between blocks, the Python Markdown block parser will consider that content as all one paragraph.

Once we parse those math equations, preserving them, and output them to the final HTML document, only then does it matter how MathJax would parse them. We don't even output MathJax with the `$$`. We normalize the format to something common depending on the output format you have [configured](https://facelessuser.github.io/pymdown-extensions/extensions/arithmatex/#mathjax-output-format).

The whole purpose of Arithmatex is to find and recognize math equations so we can prevent the Markdown parser from making a mess of them. Then you can use MathJax in the final HTML to convert them to actual math equations.

**Comment by memeplex** [comment]
Great, thanks for the clarification. BTW, I discovered your extension suite today and it's fantastic, extremely full featured and well documented, thanks!

**Comment by facelessuser** [comment]
No problem. Glad you find them useful 🙂.

**Comment by memeplex** [comment]
Sorry to bother you again, but just out of curiosity why:

> When using MathJax, for best results, it is advised to not use generic mode, and configure MathJax without the text2jax extension since MathJax automatically detects Arithmatex's default output.

?

**Comment by facelessuser** [comment]
At the time, I believe it was the recommended way from MathJax, IIRC. I think that may have changed with the 3.X series. Regardless, it is an output that is hard to mess up parsing with as all math is contained within script elements. The other methodology requires using plain text patterns, which could give false positives.
